learning_resources_search/api.py

```diff
@@ -394,7 +394,18 @@ def generate_aggregation_clause(
     current_path = ".".join(path_pieces[0:_current_path_length])
 
     if current_path == path:
-        return {"terms": {"field": path, "size": 10000}}
+        bucket_agg = {"terms": {"field": path, "size": 10000}}
+        if _current_path_length == 1:
+            return bucket_agg
+        else:
+            # In case of nested aggregations, use reverse_nested to return the
+            # root document count to avoid overcounting. For example, a resource
+            # with 5 runs all with level high_school would otherwise count 5
+            # times toward a level aggregation.
+            #
+            # Strictly speaking, this is only necessary for fields that may
+            # contain arrays with duplicated field values.
+            return {**bucket_agg, "aggs": {"root": {"reverse_nested": {}}}}
```

Comment:
This added `reverse_nested` root document count to all nested aggregations.

I don't think we actually always need it. We need it for levels because resources can have multiple runs with the same level. But for some nested fields (like platform, offered_by) resources always have exactly 1 platform / offerer.

I also considered adding `reverse_nested` as a configurable property on `FilterConfig` in LEARNING_RESOURCE_SEARCH_FILTERS. If someone thinks that's better, happy to make the change, but this seemed simpler.